pyfolio/plotting.py

```diff
@@ -641,7 +649,25 @@ def plot_rolling_returns(
             ax.fill_between(cone_df_live.index,
                             cone_df_live.sd_down,
                             cone_df_live.sd_up,
-                            color='red', alpha=0.30)
+                            color='steelblue', alpha=0.25)
+
+            if type(cone_std) is list:
+                for cone_i in list(range(1, len(cone_std))):
+                    cone_df = timeseries.cone_rolling(
+                        returns,
+                        num_stdev=cone_std[cone_i],
+                        cone_fit_end_date=live_start_date)
+
+                    cone_df_fit = cone_df[cone_df.index < live_start_date]
+
+                    cone_df_live = cone_df[cone_df.index > live_start_date]
+                    temp_keep_days = cone_df_live.index < returns.index[-1]
+                    cone_df_live = cone_df_live[temp_keep_days]
+
+                    ax.fill_between(cone_df_live.index,
+                                    cone_df_live.sd_down,
+                                    cone_df_live.sd_up,
+                                    color='steelblue', alpha=0.25)
```

Comment:
I basically did the quickest implementation to touch the least amount of code so yea it can definitely be cleaned up further. I figured it's not that computationally expensive so went to simple rather than elegant route

**Sent from iPhone**




On Tue, Oct 13, 2015 at 1:45 PM -0700, "Thomas Wiecki" <notifications@github.com> wrote:












In pyfolio/plotting.py:
> +                for cone_i in list(range(1, len(cone_std))):
> +                    cone_df = timeseries.cone_rolling(
> +                        returns,
> +                        num_stdev=cone_std[cone_i],
> +                        cone_fit_end_date=live_start_date)
> +
> +                    cone_df_fit = cone_df[cone_df.index < live_start_date]
> +
> +                    cone_df_live = cone_df[cone_df.index > live_start_date]
> +                    temp_keep_days = cone_df_live.index < returns.index[-1]
> +                    cone_df_live = cone_df_live[temp_keep_days]
> +
> +                    ax.fill_between(cone_df_live.index,
> +                                    cone_df_live.sd_down,
> +                                    cone_df_live.sd_up,
> +                                    color='steelblue', alpha=0.25)



I would take this whole logic and put it into a local function (def draw_cone()) that gets called inside the loop.



—
Reply to this email directly or view it on GitHub.